### PR TITLE
Feat(trino): add support for the ON OVERFLOW clause in LISTAGG

### PR DESCRIPTION
**Pull request by georgesittas**
See #4337

### CODE REVIEW

**Comment by VaggelisD** [comment]
Seems like `ruff` was bumped, is the investigation around it finished? 

PS: I thought it was pinned by us, is it not?

**Comment by georgesittas** [comment]
> Seems like `ruff` was bumped, is the investigation around it finished?
> 
> PS: I thought it was pinned by us, is it not?

Yeah, there's no problem bumping it. The reason it was pinned in the first place was to avoid random CI breakages.

File: sqlglot/dialects/bigquery.py [hunk 1]
```diff
@@ -1049,3 +1049,13 @@ def version_sql(self, expression: exp.Version) -> str:
...
+
+        @generator.unsupported_args("group", "parameters")
+        def regexpextract_sql(self, e: exp.RegexpExtract) -> str:
+            return self.func(
+                "REGEXP_EXTRACT",
+                e.this,
+                e.expression,
+                e.args.get("position"),
+                e.args.get("occurrence"),
+            )
```
Comment (VaggelisD) [inline sqlglot/dialects/bigquery.py:1061]: This should have been removed after the latest rebase

**Review by VaggelisD — APPROVED** [review]

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -5765,8 +5765,14 @@ class Greatest(Func):
     is_var_len_args = True
 
 
+# Trino's `ON OVERFLOW TRUNCATE [filler_string] {WITH | WITHOUT} COUNT`
+# https://trino.io/docs/current/functions/aggregate.html#listagg
+class OverflowTruncateBehavior(Expression):
+    arg_types = {"this": False, "with_count": True}
```
[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlglot/expressions.py:5771]: The latter, I didn't feel like adding more args to `GroupConcat` in order to represent the overflow behavior spec.

File: sqlglot/dialects/bigquery.py (hunk 1 above)
Comment (georgesittas) [inline sqlglot/dialects/bigquery.py:1061]:
```suggestion
```

**Pull request merged**